joint_teapot/utils/joj3.py

```diff
@@ -0,0 +1,102 @@
+import csv
+import json
+import os
+from datetime import datetime
+from typing import Any, Dict
+
+from joint_teapot.utils.logger import logger
+
+
+def generate_scoreboard(score_file_path: str, scoreboard_file_path: str) -> None:
+    if not scoreboard_file_path.endswith(".csv"):
+        logger.error(
+            f"Scoreboard file should be a .csv file, but now it is {scoreboard_file_path}"
+        )
+        return
+
+    # Load the csv file if it already exists
+    if os.path.exists(scoreboard_file_path):
+        with open(scoreboard_file_path, newline="") as file:
+            reader = csv.reader(file)
+            rows = list(reader)
+        columns = rows[0]
+        data = rows[1:]
+    else:
+        columns = [
+            "",
+            "last_edit",  # FIXME:
+            # This is just to make changes in the file so that it can be pushed.
+            # Only used in development stage. Will be removed in the future.
+            "total",
+        ]
+        data = []
+
+    column_updated = [False] * len(columns)  # Record wether a score has been updated
+    # Update data
+    with open(score_file_path) as json_file:
+        scorefile: Dict[str, Any] = json.load(json_file)
+
+    student = f"{scorefile['studentname']} {scorefile['studentid']}"
```

Comment:
use the current scorefile format, make name and id parameters to this function, and also cli args in `app.py`.

Reply:
In case there are different key names? Will we really meet such cases?

Reply:
Oh I think I get your idea